src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -455,7 +456,7 @@ public async Task<ImmutableArray<string>> GetF1KeywordsAsync(CancellationToken c
         ErrorHandler.ThrowOnFailure(vsView.GetBuffer(out var textLines));
         ErrorHandler.ThrowOnFailure(textLines.GetLanguageServiceID(out var languageServiceGuid));
 
-        var languageService = await ((AsyncServiceProvider)AsyncServiceProvider.GlobalProvider).QueryServiceAsync(languageServiceGuid).WithCancellation(cancellationToken);
+        var languageService = await new AsyncServiceProvider((COMAsyncServiceProvider.IAsyncServiceProvider)AsyncServiceProvider.GlobalProvider).QueryServiceAsync(languageServiceGuid).WithCancellation(cancellationToken);
```

Comment:
As this is a fix for a downcast that wasn't documented to succeed, maybe your fix should avoid doing the same thing again. I believe the only supported way to get the COM IAsyncServiceProvider is by querying for it.

Reply:
Querying for it is easy enough. Do you know which service interface matches it?

Reply:
There isn't a different service interface for it. So it would look like this:

```cs
AsyncServiceProvider.GlobalProvider.GetService<COMAsyncServiceProvider.IAsyncServiceProvider>()
```

You'll have to test it though. This is all based on my recollection.

Reply:
➡️ I'll submit it as a separate follow-up PR since the current one already passed integration tests.

Reply:
Yeah that will "work" as the managed version implements the native interface, but the right change is to query it as suggested by Andrew.

Reply:
➡️ Submitted #74100